docs/docs/plugins/02-dev-plugins.md

```diff
@@ -42,31 +42,31 @@ which implies to implement a predefined interface:
 // An ignite plugin must implements the Plugin interface.
 type Interface interface {
 	// Manifest declares the plugin's Command(s) and Hook(s).
-	Manifest() (Manifest, error)
+	Manifest(context.Context) (*Manifest, error)
 
 	// Execute will be invoked by ignite when a plugin Command is executed.
 	// It is global for all commands declared in Manifest, if you have declared
 	// multiple commands, use cmd.Path to distinguish them.
-	Execute(cmd ExecutedCommand) error
+	Execute(context.Context, *ExecutedCommand) error
```

Comment:
Just to check, do we need pointers here for the command?

Reply:
Proto generated types have their inner types generated as pointers, in this case it is a pointer because of the request types user for gRPC. I will double check to see if its possible to at least remove the pointer from the interface.

Reply:
We must use pointers for all generated types, one of the reasons is that they contain a private `state` field that have a `sync.Mutex`. This mutex is added intentionally to avoid shallow copies.